pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -396,14 +389,20 @@ def post_dispatch(self, trainer: "pl.Trainer") -> None:
 
     def barrier(self, *args, **kwargs) -> None:
         if not distributed_available():
-            return
+            raise RuntimeError("DDP is not initialized and torch.distributed is not avalible, can not broadcast object")
         if _TORCH_GREATER_EQUAL_1_8 and torch.distributed.get_backend() == "nccl":
             torch.distributed.barrier(device_ids=self.determine_ddp_device_ids())
         else:
             torch.distributed.barrier()
 
     def broadcast(self, obj: object, src: int = 0) -> object:
-        return self.dist.broadcast(obj)
+        if not distributed_available():
+            raise RuntimeError("DDP is not initialized and torch.distributed is not avalible, can not broadcast object")
```

Comment:
this should throw automatically as part of  torch distributed's broadcast implementation